test/integration/issuance_test.go

```diff
@@ -167,3 +170,69 @@ func TestIssuanceProfiles(t *testing.T) {
 	test.AssertEquals(t, len(legacy.SubjectKeyId), 20)
 	test.AssertEquals(t, len(modern.SubjectKeyId), 0)
 }
+
+// TestIPShortLived verifies that we will allow IP address identifiers only in
+// orders that use the shortlived profile.
+func TestIPShortLived(t *testing.T) {
+	t.Parallel()
+
+	// Create an account.
+	client, err := makeClient("mailto:example@letsencrypt.org")
+	if err != nil {
+		t.Fatalf("creating acme client: %s", err)
+	}
+
+	_, shortlived := client.Directory().Meta.Profiles["shortlived"]
```

Comment:
It feels odd to include this check, but not similar checks for the "legacy" and "modern" profiles, too. Either check all three, or check none. I think checking none is fine, because order rejection due to unknown profile is a specific error message that can be detected.